### ISSUE TITLE
Relationship lookup component displays misleading information

### ISSUE THREAD

**Issue by jlipka** [DSpace/dspace-angular#3312]
## Describe the bug
There is a bug in DSpace when you are in the submission process and use the lookup to search for an author to create a new relationship. The information displayed in the component is incorrect and does not reflect the current relationship state of the submission item.

_My test was made locally with DSpace in version 8, the same error occurs in the sandbox (sandbox.dspace.org) with Version 9._

## To reproduce
1. Login (using an account/role that is allowed to create any new submission)
2. Create a new item in any collection that contains an author lookup (e.g., "Publications > Publications 2").
3. Open the author lookup modal by clicking on the search icon.
4. *(implicit: some authors are listed automatically; when testing with a new submission, they are unchecked)*.
5. Start typing in the search input field and enter a string that will certainly not return any results, e.g. "asdf".
6. Click the arrow next to " Person Search Results" and click "Select Page".

Now there are several problems to the user:
1. The still visible "Current Selection (X)" tab indicates that there are 5 (example) active relationships, even if you click inside the tab, it seems like multiple entities are chosen as a relation
2. If you close the lookup modal the authors-list of the submission form has been extended by only one author (the first of the initial search result)
3. if you click on "Add more", start a new (empty) search and the modal opens once again, the "Current selection" tab still indicates the wrong list of active relationships.
4. if you click on "Add more", start a new search using a nonsense query (e.g. "asdf") again. The modal opens, and after clicking on "Select page" an error occurs (see in browser console); the error seems to break the functionality of the component and any further search will lead in an endless loading animation.

## Expected behavior
1. The drop-down menu next to the search results to select all items should only appear if there is a list of results.
2. The functionality behind "Select page" / "Deselect page" should only be based on the current search results, not on any old data from any previous/initial search.

## Screenshots
![2024-09-09_17-40-author-lookup-1](https://github.com/user-attachments/assets/cc0a1342-06ff-482e-b415-c732f5105e8b)

![2024-09-09_17-40-author-lookup-2](https://github.com/user-attachments/assets/63151a69-594e-48cf-87fe-d57e894a0c3b)

![2024-09-09_17-40-author-lookup-3](https://github.com/user-attachments/assets/4386c5d7-da62-4ed0-8c76-4108b064f8ba)

![2024-09-09_17-40-author-lookup-4](https://github.com/user-attachments/assets/d09e6272-7305-4228-b463-86b3e6bb2fd9)

![2024-09-09_17-40-author-lookup-5](https://github.com/user-attachments/assets/0b823f9c-e372-4bd4-8ae3-ff4cba352301)

**Comment by jlipka** [comment]
> If I'm understanding correctly, the core issue here is that that arrow dropdown (to the right of "Person Search Results") should **not be displayed** when there are no results.

Yes, I think you're right - your proposed change should solve all the problems described.

**Comment by Andrea-Guevara** [comment]
Good morning @tdonohue!

I've been looking for a few days and testing possible solutions to this problem and I've come up with a few possibilities:

1. Put the following conditional with *ngIf in the select/deselect dropdown of the page: 
*ngIf="(resultsRD$ | async)?.page.length > 0”

2. Put a disable property on the button, so that if the search returns no results, the select/deselect options are disabled. This would take advantage of the variable “*ngVar=”(resultsRD$ | async) as resultsRD” : 
[disabled]=”!resultsRD?.page.length > 0” 

3. Make the “selectPage()” method only run when the page has an item on it:
selectPage(page: SearchResult<DSpaceObject>[]) {
    if (!page || page.length === 0) {
      return;
    }
    this.selection$
      .pipe(take(1))
      .subscribe((selection: SearchResult<Item>[]) => {
        const filteredPage: SearchResult<DSpaceObject>[] = page.filter((pageItem: SearchResult<DSpaceObject>) => selection.findIndex((selected: SearchResult<Item>) => selected.equals(pageItem)) < 0);
      if (filteredPage.length > 0) {
        this.selectObject.emit(...filteredPage);
      }
    });
    this.selectableListService.select(this.listId, page);
  }

I've tested all these different ways and it works, but unfortunately the condition is only reproduced once and then it doesn't work anymore. The tests were carried out on the component:
src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/search-tab

I hope this helps in some way, and if you have any suggestions, please let me know.

**Comment by Andrea-Guevara** [comment]
Good afternoon @tdonohue, I hope you're well!

I didn't explain myself very well!
I meant that my attempts to solve this problem only work once. It seems that some cache is stored and the conditionals are not able to identify the changes. For example:

If I open the modal and it has results the return would be “true” but if no results are present, the return would be “false”. It seems that this switch between true and false isn't identified, so it's almost impossible to condition the dropdown so that it doesn't appear if no results are found.

I hope I've made myself clear haha. If it's still not clear, let me know.

**Comment by tdonohue** [comment]
@Andrea-Guevara : Ok, I understand now.

Unfortunately, I'm not sure of the answer myself.  It does sound like something is cached here.  It may need some digging / debugging to understand what is going on.  That's not something I have time for at the moment.   

The only alternative is to find a different way to solve this issue.  Obviously, the core issue here is that you shouldn't be able to select a page if there are no pages.  I'm not sure myself of a better solution at the moment.  Hopefully, if another developer has an idea here, they'll chime in on this discussion.

**Comment by jlipka** [comment]
@Andrea-Guevara Thank you for your initial research and solutions. I have taken your suggestions and discussion and submitted a PullRequest proposal. There you can also see the possible solution that an event is still triggered if the result set is empty - I think that was still the problem?
I look forward to your feedback, or possibly a merge if everything is OK.
